### PR TITLE
OW1 doomfist reenabled

### LINKED ISSUE #39: Revert reworked Doomfist to OW1

### PR DESCRIPTION
**Pull request by snappycreeper**
Modified health and abilities to mimic OW1 doomfist. Resolves #39 

Doomfist's slam and uppercut keybinds are swapped for now.

### CODE REVIEW

File: src/base_settings.opy [hunk 1]
```diff
@@ -116,6 +116,14 @@ settings {
...
                 "ultGen%": percent(ow2_ult_cost_REAPER/ow1_ult_cost_REAPER)
             },
+            "doomfist": {
+                "ammoRegenerationTime%": 135,
+                "ultKb%": 300,
+                "ability1ChargeRate%": 10,
+                "secondaryFireKb%": 115,
+                "ultDuration%": 150,
+                "ultGen%": percent(ow2_ult_cost_DOOMFIST/ow1_ult_cost_DOOMFIST)
+            },
```
Comment (MaxwellJung) [inline src/base_settings.opy:126]: Can you move these lines up so the heroes are sorted alphabetically?

File: src/custom_heroes.opy [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 #!include "heroes/ana.opy"
 #!include "heroes/brigitte.opy"
+#!include "heroes/doomfist.opy"
```
Comment (MaxwellJung) [inline src/custom_heroes.opy:10]: Same comment as before

The imports should be grouped by tank, damage, and support heroes (+ alphabetic for each group).

File: src/heroes/doomfist.opy [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    eventPlayer.setStatusEffect(eventPlayer, Status.UNKILLABLE, 9999)
+
+
+rule "Doom shields":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.getHealthOfType(Health.SHIELDS) < 100
+    
+    eventPlayer.clearStatusEffect(Status.UNKILLABLE)
+
```
Comment (MaxwellJung) [inline src/heroes/doomfist.opy:18]: What's the purpose of these rules?

File: src/heroes/doomfist.opy [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    
+    eventPlayer.applyImpulse(Vector.UP, 15, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).applyImpulse(Vector.UP, 15, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)
+    #getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setStatusEffect(eventPlayer, Status.STUNNED, 1.25)
+    damage(getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF), eventPlayer, 50)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setMoveSpeed(0)
+    eventPlayer.cancelPrimaryAction()
+    wait(0.95)
+    eventPlayer.setGravity(0)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setStatusEffect(eventPlayer, Status.ROOTED, 0.6)
```
Comment (MaxwellJung) [inline src/heroes/doomfist.opy:60]: Maybe turning LOSCheck On might fix #69

File: src/heroes/doomfist.opy [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+
+rule "DOOMS FIST":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.isFiringSecondaryFire() == true
+    
+    eventPlayer.setDamageDealt(370)
+    wait(0.15)
+    eventPlayer.setDamageDealt(370)
```
Comment (MaxwellJung) [inline src/heroes/doomfist.opy:74]: Magic numbers; should be replaced with keywords from ow1/ow2_constants.opy. I'll fix these magic numbers myself if you don't want to do it.

**Review by MaxwellJung — CHANGES_REQUESTED** [review]
I'll merge this to my doomfist branch after you fix the changes requested.
You can expect doomfist -> staging -> main for v0.0.7.

File: src/custom_heroes.opy (hunk 2 above)
Comment (snappycreeper) [reply src/custom_heroes.opy:10]: Yes, ill do that

File: src/heroes/doomfist.opy (hunk 3 above)
Comment (snappycreeper) [reply src/heroes/doomfist.opy:18]: I just had them when I was testing, I forgot to remove them

File: src/heroes/doomfist.opy (hunk 4 above)
Comment (snappycreeper) [reply src/heroes/doomfist.opy:60]: Will try that

Comment (snappycreeper) [reply src/heroes/doomfist.opy:60]: Minor bug, don't think it can be fixed

File: src/heroes/doomfist.opy (hunk 5 above)
Comment (snappycreeper) [reply src/heroes/doomfist.opy:74]: Ill fix this later

**Review by MaxwellJung — APPROVED** [review]

**Pull request merged**